backend/src/socket/socket.gateway.ts

```diff
@@ -34,242 +36,186 @@ export class SocketGateway implements OnGatewayConnection {
+  // INVITE
+  /****************************************************************************/
+
+  @SubscribeMessage('inviteUserToPlay')
+  async handleInvitePlayer(client: Socket, payload: string) {
+    // find and check the "invited" user
+    const invitedUserEmail: string = payload;
+    const user: User = await this.gameService.getSocketUser(client);
+    if (!invitedUserEmail || invitedUserEmail === '') {
       this.server
         .to(client.id)
-        .emit('errorGameInvite', { error: 'Player not found' });
-      return;
+        .emit('errorGameInvite', { error: 'User Email not provided.' });
+      return ;
     }
-
-    console.log(invitedPlayer);
-    if (invitedPlayer.status !== 'ONLINE') {
-      console.log('not available', user);
+    const invitedUserStatus: string = await this.gameService.checkInvitedUserStatus(client, invitedUserEmail);
+    if (invitedUserStatus === 'ONLINE') {
       this.server
         .to(client.id)
-        .emit('errorGameInvite', { error: 'Player not available' });
+        .emit('invitationSent', { invitedUserEmail: payload });
+    } else {
+      this.server
+        .to(client.id)
+        .emit('errorGameInvite', { error: invitedUserStatus });
       return;
     }
 
-    this.gameService.createWaitingGame(currentPlayer);
-    console.log('player before emit', currentPlayer);
-    const gameID = currentPlayer.gameID.toString();
-    client.join(gameID);
-    this.server
-      .to(invitedPlayer.socketID)
-      .emit('gameInvite', { invitedBy: currentPlayer.userName });
+    // create the game and wait
+    console.log("create inviting game and wait")
+    const invitingPlayer: Player = await this.gameService.createPlayer(client);
+    const invitingGame: Game = await this.gameService.createInvitingGame(invitingPlayer);
+    console.log(invitingGame);
+    if (invitingGame) {
+      client.join(invitingGame.gameID);
+      const invitedUserSocketID: string = await this.cacheManager.get(`game${invitedUserEmail}`);
+      this.server
+        .to(invitedUserSocketID)
+        .emit('gameInvite', { invitedBy: invitingPlayer.email});
+    }
   }
 
-  @SubscribeMessage('respondToInvite')
-  async handleRespondToInvite(client: Socket, payload: string) {
-    const response: { accept: boolean; invitedBy: string } =
-      JSON.parse(payload);
-    const currentPlayer: Player = this.clients.find(
-      (c) => c.socketID === client.id,
-    );
-    const invitingPlayer: Player = this.clients.find(
-      (c) => c.userName === response.invitedBy,
-    );
+  // Make accept/decline seperately so the payload can be a simple string
 
+  @SubscribeMessage('acceptInvitation')
+  async handleAcceptInvitation(client: Socket, payload: string) {
+    // create the invitedPlayer
+    const invitedPlayer: Player = await this.gameService.createPlayer(client);
+    const invitingPlayer: Player = await this.gameService.getPlayerByEmail(payload);
     if (!invitingPlayer) {
```

Comment:
And maybe check if their status is still 'ONLINE'. They might have sent an invitation and then started a random game in the mean time.